ADALiOS/ADALiOS/ADAuthenticationContext+WebRequest.m

```diff
@@ -295,9 +296,9 @@ - (NSString*)queryStringFromResource:(NSString*)resource
     
     [startUrl appendFormat:@"&%@", [[ADLogger adalId] adURLFormEncode]];
     
-    if (![NSString adIsStringNilOrBlank:userId])
+    if (![NSString adIsStringNilOrBlank:userId.userId])
     {
-        [startUrl appendFormat:@"&%@=%@", OAUTH2_LOGIN_HINT, [userId adUrlFormEncode]];
+        [startUrl appendFormat:@"&%@=%@", OAUTH2_LOGIN_HINT, [userId.userId adUrlFormEncode]];
```

Comment:
In .NET it appears no login hint is passed in at all if it's not a displayableId. I will copy that behavior.